### ISSUE TITLE
uninitialized matrices in osqp causing a crash

### ISSUE THREAD

**Issue by samuelstjean** [stephane-caron/qpsolvers#10]
Seems like there is a problem in passing None matrices to some solvers, for example

this works
~~~python
from qpsolvers import solve_qp 
#maybe solve_safer_qp needs checking, no idea on this one


solve_qp(np.eye(10), np.ones(10), solver='quadprog') # it works only in master though, not in the pypi release as it was fixed afterwards)
solve_qp(np.eye(10), np.ones(10), solver='ecos')
~~~

this does not
~~~python
solve_qp(np.eye(10), np.ones(10), solver='osqp')
---------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1977-b6d562d21732> in <module>()
      1 solve_qp(np.eye(10), np.ones(10), solver='quadprog')
----> 2 solve_qp(np.eye(10), np.ones(10), solver='osqp')
      3 solve_qp(np.eye(10), np.ones(10), solver='ecos')

~/anaconda3/lib/python3.5/site-packages/qpsolvers/__init__.py in solve_qp(P, q, G, h, A, b, solver, initvals, sym_proj)
    168         return mosek_solve_qp(P, q, G, h, A, b, initvals=initvals)
    169     elif solver == 'osqp':
--> 170         return osqp_solve_qp(P, q, G, h, A, b, initvals=initvals)
    171     elif solver == 'qpoases':
    172         return qpoases_solve_qp(P, q, G, h, A, b, initvals=initvals)

~/anaconda3/lib/python3.5/site-packages/qpsolvers/osqp_.py in osqp_solve_qp(P, q, G, h, A, b, initvals)
     74     <https://github.com/oxfordcontrol/osqp/issues/10>`_ in your solutions.
     75     """
---> 76     l = -inf * ones(len(h))
     77     if type(P) is ndarray:
     78         warn(conversion_warning("P"))

TypeError: object of type 'NoneType' has no len()
~~~
Looks like it is due to this line

https://github.com/stephane-caron/qpsolvers/blob/7d71cdfe5657ca098a4c6c9ff2b6627956d7631d/qpsolvers/osqp_.py#L76

Because we never initialize h, the inner wrapper function should probably do it

**Comment by stephane-caron** [comment]
Thank you for pointing this out :) There are indeed holes in the API. I've added some systematic tests in [test_solvers.py](https://github.com/stephane-caron/qpsolvers/blob/master/examples/test_solvers.py) that reproduce this error (and raises some others).

PS: thanks also for pointing out the pypi discrepancy. I've released 1.0.6 to push the fixes, and will release 1.0.7 once we are done with this issue.
